cli/cmd/check.go

```diff
@@ -45,6 +46,7 @@ func (options *checkOptions) nonConfigFlagSet() *pflag.FlagSet {
 	flags.BoolVar(&options.cniEnabled, "linkerd-cni-enabled", options.cniEnabled, "When running pre-installation checks (--pre), assume the linkerd-cni plugin is already installed, and a NET_ADMIN check is not needed")
 	flags.StringVarP(&options.namespace, "namespace", "n", options.namespace, "Namespace to use for --proxy checks (default: all namespaces)")
 	flags.BoolVar(&options.preInstallOnly, "pre", options.preInstallOnly, "Only run pre-installation checks, to determine if the control plane can be installed")
+	flags.BoolVar(&options.cniOnly, "cni", options.cniOnly, "Only run CNI checks, to determine if the plugin is installed and ready")
```

Comment:
Can I get a high level description of the checks that are added in this PR?

Do these just check for the existence of certain resources that need to exist for CNI to work?  If so, shouldn't we always do those checks if the --linkerd-cni-enabled flag is set?  I don't understand the purpose of the `--cni` flag compared to `--linkerd-cni-enabled`.  Why does this flag disable other checks?

Reply:
Yes this actually got a bit more confusing that it needs to. I will revise it. Otherwise the just of the checks are: 

- Check for the resources that need to exist
- Check for the state of the deamonset (i.e. how many nodes are in the cluster vs how many nodes have the cni  pod up and running)